### PR TITLE
Add generator tests

### LINKED ISSUE #718: Unit tests for sig docs generator
We have a cool new generator for sig docs, but there's no go unit tests for it.

cc: @jamiehannaford @jbeda

### PR DESCRIPTION
**Pull request by jamiehannaford**
Fixes #718. Current coverage is ~87%, which I think is enough for a first attempt.

cc/ @cblecker

### CODE REVIEW

**Comment by jamiehannaford** [comment]
@cblecker would you mind taking a look, or pinging relevant folks? Thanks! :)

**Comment by cblecker** [comment]
@jamiehannaford needs rebase from removal of timestamps

**Comment by jamiehannaford** [comment]
@cblecker Ping

**Comment by cblecker** [comment]
@jamiehannaford I'll take a look at these early next week (unless someone else is able to review them sooner)

File: generator/Dockerfile [hunk 1]
```diff
@@ -1 +1 @@
-FROM golang:1.6-onbuild
+FROM golang:1.8-onbuild
```
Comment (cblecker) [inline generator/Dockerfile]: onbuild is deprecated. let's use the 1.8 image directly, but then add the wrapper commands to our dockerfile:
https://github.com/docker-library/golang/blob/132cd70768e3bc269902e4c7b579203f66dc9f64/1.8/onbuild/Dockerfile

File: generator/testdata/custom_content.md [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+
+## Contact
+* [Slack](https://kubernetes.slack.com/messages/sig-auth)
+* [Mailing list](https://groups.google.com/forum/#!forum/kubernetes-sig-auth)
+
+<!-- BEGIN CUSTOM CONTENT -->
+FOO BAR BAZ
+<!-- END CUSTOM CONTENT -->
+
+Last generated:  Wed Jun 7 2017 14:27:56
```
Comment (cblecker) [inline generator/testdata/custom_content.md]: Generated date needs to be removed from testdata

File: generator/testdata/no_custom_content.md [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+
+## Contact
+* [Slack](https://kubernetes.slack.com/messages/sig-auth)
+* [Mailing list](https://groups.google.com/forum/#!forum/kubernetes-sig-auth)
+
+<!-- BEGIN CUSTOM CONTENT -->
+
+<!-- END CUSTOM CONTENT -->
+
+Last generated:  Wed Jun 7 2017 14:27:56
```
Comment (cblecker) [inline generator/testdata/no_custom_content.md]: Generated date needs to be removed from testdata

File: generator/app.go [hunk 4]
```diff
@@ -213,71 +203,33 @@ func writeCustomContentBlock(f *os.File, content string) {
...
+		group.SetupGitHubTeams(prefix)
 
-		for _, gtn := range githubTeamNames {
-			wg.Contact.GithubTeamNames = append(wg.Contact.GithubTeamNames, fmt.Sprintf("%s-%s", prefix, gtn))
-		}
-
-		outputPath := fmt.Sprintf("%s/%s", dirName, sigIndexOutput)
-		if err := writeTemplate(wgIndexTemplate, outputPath, wg); err != nil {
+		outputPath := filepath.Join(group.Dir, indexFilename)
+		if err := writeTemplate(readmeTemplate, outputPath, group); err != nil {
```
Comment (cblecker) [inline generator/app.go]: `2017/07/06 18:33:42 open /go/src/app/readme.tmpl: no such file or directory`

File: Makefile [hunk 5]
```diff
@@ -13,3 +13,7 @@ gen-doc:
 
 gen-docs:
 	docker run -v $(shell pwd):/go/src/app sigdocs
+
+test-unit:
```
Comment (cblecker) [inline Makefile]: It's good to have a dockerized path, but a regular `go test ./generator/...` should work too. it currently leaves a dirty path.

File: Makefile [hunk 6]
```diff
@@ -13,3 +13,7 @@ gen-doc:
...
+	docker build -t sigdocs-test -f generator/Dockerfile.test generator
+	docker run sigdocs-test
```
Comment (cblecker) [inline Makefile]: We can likely rm the running container after completion. No need to leave cruft around.

**Comment by jamiehannaford** [comment]
Thanks for the feedback @cblecker! I just implemented all your recommendations, PTAL when you get some time

File: generator/.gitignore [hunk 7]
```diff
@@ -0,0 +1 @@
+sigs.yaml
```
Comment (cblecker) [inline generator/.gitignore]: With the changes, I think this should be `generated/` to ignore that whole directory.

File: Makefile [hunk 8]
```diff
@@ -9,7 +9,11 @@ build-image:
...
 
 gen-doc:
-	docker run -e WG=${WG} -e SIG=${SIG} -v $(shell pwd):/go/src/app sigdocs
+	docker run -e WG -e SIG -v $(shell pwd):/go/src/app sigdocs
 
 gen-docs:
-	docker run -v $(shell pwd):/go/src/app sigdocs
+	docker run -v $(shell pwd):/go/src/app/generated sigdocs
+
+test-unit:
```
Comment (cblecker) [inline Makefile]: Can we change this target to `test`? It aligns it with `make test` in k/k

File: Makefile [hunk 9]
```diff
@@ -9,7 +9,11 @@ build-image:
 	docker build -t sigdocs -f generator/Dockerfile generator
 
 gen-doc:
-	docker run -e WG=${WG} -e SIG=${SIG} -v $(shell pwd):/go/src/app sigdocs
+	docker run -e WG -e SIG -v $(shell pwd):/go/src/app sigdocs
 
 gen-docs:
-	docker run -v $(shell pwd):/go/src/app sigdocs
+	docker run -v $(shell pwd):/go/src/app/generated sigdocs
```
Comment (cblecker) [inline Makefile]: I wonder if we should `--rm` these after too?

File: Makefile [hunk 10]
```diff
@@ -9,7 +9,11 @@ build-image:
 	docker build -t sigdocs -f generator/Dockerfile generator
```
Comment (cblecker) [inline Makefile]: I wonder if we can give this a more descriptive image name? Like perhaps kube-communitydocs ?

File: Makefile [hunk 11]
```diff
@@ -9,7 +9,11 @@ build-image:
...
 gen-doc:
-	docker run -e WG=${WG} -e SIG=${SIG} -v $(shell pwd):/go/src/app sigdocs
+	docker run -e WG -e SIG -v $(shell pwd):/go/src/app sigdocs
 
 gen-docs:
-	docker run -v $(shell pwd):/go/src/app sigdocs
+	docker run -v $(shell pwd):/go/src/app/generated sigdocs
+
+test-unit:
+	docker build -t sigdocs-test -f generator/Dockerfile.test generator
```
Comment (cblecker) [inline Makefile]: I'm also wondering if we actually need a separate image for testing, or if we could use the same docs image, but then override the CMD from docker run.

File: generator/.gitignore (hunk 7 above)
Comment (jamiehannaford) [inline generator/.gitignore]: Good spot 👍

File: Makefile (hunk 11 above)
Comment (jamiehannaford) [inline Makefile]: Yeah, agreed. No need for 2 separate ones considering the delta is so small now

**Comment by jamiehannaford** [comment]
@cblecker Latest round of changes are up, thanks again for the feedback.

BTW, since this is code, is it worth having an OWNERS file in the `generator` dir? Or is it not worth the bother?

File: generator/.gitignore [hunk 12]
```diff
@@ -0,0 +1 @@
+generated/sigs.yaml
```
Comment (cblecker) [inline generator/.gitignore]: This should just be:
```
generated/`
```
so it ignores that whole directory.

File: generator/Dockerfile.test [hunk 13]
```diff
@@ -0,0 +1,9 @@
+FROM golang:1.8
```
Comment (cblecker) [inline generator/Dockerfile.test]: We don't need this file anymore

**Comment by cblecker** [comment]
Oh, and on the topic of OWNERS, I'm not really sure. We don't have bots running in kubernetes/community to use them, so I don't think they bring much value to the table at the moment.

**Comment by cblecker** [comment]
Thanks @jamiehannaford !

**Pull request merged**